x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamInput.java

```diff
@@ -180,6 +193,56 @@ public EsqlConfiguration configuration() throws IOException {
         return configuration;
     }
 
+    @Override
+    public int readArraySize() throws IOException {
+        return super.readArraySize();
+    }
+
+    /**
+     * Read a {@link Block} as part of the plan.
+     * <p>
+     *     These {@link Block}s are not tacked by {@link BlockFactory} and closing them
+     *     does nothing so they should be small. We do make sure not to send duplicates,
+     *     reusing blocks sent as part of the {@link EsqlConfiguration#tables()} if
+     *     possible, otherwise sending a {@linkplain Block} inline.
+     * </p>
+     */
+    public Block readBlock() throws IOException {
```

Comment:
Could we rename this method and its counterpart to something else, like `readCachedBlock`? I'm worried that these methods can be misused.

Reply:
++

I was calling it that for a while then I stopped. Not sure why.